src/modules/models/models.service.ts

```diff
@@ -1,14 +1,79 @@
 import { Injectable } from '@nestjs/common';
-import {ModelsRepositoryService} from './models-repository.service';
+import { ModelsRepositoryService } from './models-repository.service';
+import { ModelItemsRepositoryService } from './model-items-repository.service';
+import { ModelItemsFieldRepositoryService } from './model-items-field-repository.service';
+import { ModelItemsRelationRepositoryService } from './model-items-relation-repository.service';
+import { EntityManager, getManager } from 'typeorm';
+import { CreateModelDto } from './dto/createModel.dto';
+import { RelationItem } from './dto/relationItem.dto';
+import { ModelItem } from './dto/modelItem.dto';
 
 @Injectable()
 export class ModelsService {
   constructor(
     private modelsRepositoryService: ModelsRepositoryService,
+    private modelItemsRepositoryService: ModelItemsRepositoryService,
+    private modelItemsFieldRepositoryService: ModelItemsFieldRepositoryService,
+    private modelItemsRelationRepositoryService: ModelItemsRelationRepositoryService,
   ) {}
 
   async getModelsCount(admin: number) {
     const count = await this.modelsRepositoryService.getCount({ admin_id: admin });
     return { count };
   }
+
+  async createModel(data: CreateModelDto, admin: number) {
+    return await getManager().transaction(async manager => {
+      const { name, connectionId, items, relations } = data;
+      const model = await this.modelsRepositoryService.createModel(name, admin, connectionId, manager);
+      const { id: modelId } = model;
+      const modelItems = await Promise.all(items.map(item => this.createModelItem(item, modelId, manager)));
+      const modelItemsMap = ModelsService.createModelItemsMapForRelations(modelItems);
+      const relationsWithModelsId = ModelsService.patchRelationsDataWithModelsId(modelItemsMap, relations);
+      await Promise.all(
+        relationsWithModelsId.map(relation =>
+          this.modelItemsRelationRepositoryService.createRelation({ ...relation, connectionManager: manager }),
+        ),
+      );
+      return {
+        id: modelId,
+        name: model.name,
+        tables: modelItems.length,
+        fields: items.reduce((acc, curr) => acc + curr.rows.length, 0),
+        users: 0,
+        uses: 0,
+        active: true,
+        created: model.created_at,
+      };
+    });
+  }
+
+  private static patchRelationsDataWithModelsId(modelItemsMap: Map<string, number>, relations: RelationItem[]) {
```

Comment:
What is private static method?

Reply:
You will not believe - static method that can be used only inside the class instance)

Reply:
1) Static methods belong to class not it's instance
2) Static methods are always public (They are supposed to be)
3) YOU use it outside instance as public!!!

Remove private pozhaluysta )

Reply:
1. I know.
2. You are wrong, static methods can be private 
![image](https://user-images.githubusercontent.com/31630903/69038322-4013da80-09f2-11ea-9f9e-d90fff065892.png)
3. I never use it outside the instance

Reply:
Ok, I've got u